docs/modules/hdfs/pages/usage-guide/upgrading.adoc

```diff
@@ -0,0 +1,85 @@
+= Upgrading HDFS
+
+IMPORTANT: HDFS upgrades are experimental, and details may change at any time
+
+HDFS currently requires a manual process to upgrade. This guide will take you through an example case, upgrading an example cluster (from our xref:getting_started/index.adoc[Getting Started] guide) from HDFS 3.3.6 to 3.4.0.
```

Comment:
Should we mention anything about not using HDFS while the upgrade happens (just because of the point in time recovery for rollback?)

Reply:
Fair point, I'm honestly not sure how HDFS behaves here. Maybe we should test that...

Reply:
Makes sense! Speaking of rollback, is this something we want to test and support as well?
I'd be fine not to do for now, in that case can we please document that we don't support rollback? ^^

Reply:
So, it seems like they differentiate between rollbacks (which roll back file data, require downtime) and downgrades (which preserves file data, possible in HA).

This PR supports downgrades, but does _not_ implement the ordering requirements to preserve HA. That would be possible to implement, but requires a few (additive) CRD changes. Rollbacks are a more complicated dance that I would break out into a separate issue.

Reply:
Added some documentation on downgrades to b24c25f43b5bfa82f4233531e62c34fa901c1c35, experimenting a bit with how complicated "proper" HA downgrades would be.

Reply:
I'll leave this open then, but feel free to mark as resolved if the experiment doesn't yield the results you'd like.

Reply:
10e5220f0ad70198196b4e429fe7400cd3cf231b